### DISCUSSION TITLE
Infolist repeatable action

### DISCUSSION THREAD

**Discussion by andrieslubbertsen** [filamentphp/filament#9256] [Help]
Hello all,

I'm looking for a solution to a problem with a filament 3.0.81 View Page. On this page, I use a RepeatableEntry with an Action for each Entry. I would like to open a modal with the details of the entry that was clicked on. 

**Question**
How can I retrieve the data of the clicked entry in the modal? 'Model $record' contains the parent record of the view (Customer in my case).

**I have a simplified representation of the issue below:**
```php
return $infolist
            ->state([
                'repeatable' => [
                    [
                        'name' => 'John Doe',
                        'email' => 'john@example.com',
                    ],
                    [
                        'name' => 'Jane Doe',
                        'email' => 'jane@example.com',
                    ],
                ],
            ])
            ->schema([
                RepeatableEntry::make('repeatable')
                    ->schema([
                        TextEntry::make('name'),
                        Actions::make([
                            Action::make('View')
                                ->form(fn() => [
                                    Forms\Components\TextInput::make('name')
                                ])
                                ->fillForm(function (Model $record): array {

                                    // How do I access the data of the clicked entry at this point?
                                    return [
                                        'name' => $record['name']
                                    ];
                                })
                        ])
                    ]),
            ]);
``` 

**Simplified representation**
![Scherm­afbeelding 2023-10-23 om 08 07 32](https://github.com/filamentphp/filament/assets/107552507/07ff8d92-c3e9-4cb1-9f07-a7d3a76184af)

**Comment by leroy** [comment]
For anyone reading this and @andrieslubbertsen ofcourse.

You should be able to inject the Action itself in the fillForm Closure and call the following:

```php

Action::make()
    ->fillForm(fn(Action $action) => Arr::get($this->data, $action->getComponent()->getStatePath()))


```

Make sure to store the state in a property of your Livewire component.
